### ISSUE TITLE
Giada crash with negative MIDI port value.

### ISSUE THREAD

**Issue by hfiguiere** [monocasual/giada#515]
**Environment**
- OS: Linux
- Giada version: 0.18.1 and 0.18.2

**Describe the bug**
My `giada.conf` had `midi_port_out` set to -2. Giada then crash on startup. I don't know how or why this happened, but it did.

**To Reproduce**
1. Set `midi_port_out` to `-2` in giada.conf
2. giada will segfault.

**Expected behavior**
No crash

**Screenshots**

**Additional context**

The stack trace.
```
gdb) where
#0  MidiOutJack::getPortName[abi:cxx11](unsigned int) (this=0x555555c5bea0, portNumber=4294967294) at RtMidi.cpp:3384
#1  0x000055555566a3a3 in RtMidiOut::getPortName[abi:cxx11](unsigned int) (portNumber=<optimized out>, this=<optimized out>)
    at /app/include/rtmidi/RtMidi.h:624
#2  giada::m::kernelMidi::getOutPortName[abi:cxx11](unsigned int) (p=<optimized out>) at src/core/kernelMidi.cpp:202
#3  0x000055555566ad0b in giada::m::kernelMidi::openOutDevice (port=<optimized out>) at src/core/kernelMidi.cpp:118
#4  0x0000555555668218 in giada::m::init::(anonymous namespace)::initMIDI_ () at src/core/init.cpp:132
#5  giada::m::init::startup (argc=1, argv=0x7fffffffddd8) at src/core/init.cpp:231
#6  0x000055555560619b in main (argc=<optimized out>, argv=<optimized out>) at src/main.cpp:53
```
`InitMIDI()` pass -2 (`int`) as read from the config to `openOutDevice()`. Then it get pass to `getOutPortName()` as an `unsigned`. And that's where it crashes...

**Comment by gvnnz** [comment]
Hey @hfiguiere , it would be great to reproduce the root cause of this: why the configuration ended up with -2 as the MIDI device. Did you happen to change MIDI device/soundcard or anything else MIDI-related between two sessions?

**Comment by hfiguiere** [comment]
I don't remember, the file was a few month old.
But that shouldn't prevent from fixing the code to sanitize the input.

**Comment by gvnnz** [comment]
See also #519 .

**Comment by gvnnz** [comment]
Fixed in 3ee283433d9d30715d36fb31f63e7a4a2070e61f.
